File: 디자인패턴/java 디자인 패턴/Mediator/Window.java
public class Window extends Participant {
    private boolean bClosed = false;
    
    public Window(Mediator mediator) {
        super(mediator);
    }

    public void Open() {
        bClosed = true;
        mediator.participantChange(this);
    }

    public void Close() {
        bClosed = false;
        mediator.participantChange(this);
    }

    public boolean isWindow() {
        return bClosed;
    }

    @Override
    public String toString() {
        if(bClosed) return "# 창문: 열림";
        else return "# 창문: 닫힘";
    }
}

File: 디자인패턴/Python 디자인 패턴/Mediator/Window.py
from Participant import*

class Window(Participant):
    def __init__(this, mediator: Mediator) -> None:
        super().__init__(mediator)
        this.bWindow = False
    
    def Open(this):
        this.bWindow = True
        this.mediator.participantChange(this)
    
    def Close(this):
        this.bWindow = False
        this.mediator.participantChange(this)
    
    def isWindow(this):
        return this.bWindow
    
    def strPrint(this):
        if(this.bWindow):return "# 창문 열림"
        return "# 창문 닫힘"
    pass